conans/client/build/msbuild.py

```diff
@@ -1,21 +1,119 @@
+import re
+
 from conans import tools
-from conans.client.build.visual_environment import VisualStudioBuildEnvironment
-from conans.client.tools.win import msvc_build_command
+from conans.client.build.visual_environment import (VisualStudioBuildEnvironment,
+                                                    vs_build_type_flags, vs_std_cpp)
+from conans.client.tools.oss import cpu_count
+from conans.client.tools.win import vcvars_command
+from conans.errors import ConanException
+from conans.util.env_reader import get_env
+from conans.util.files import tmp_file
 
 
 class MSBuild(object):
 
     def __init__(self, conanfile):
-        self._conanfile = conanfile
-        self.build_env = VisualStudioBuildEnvironment(self._conanfile)
+        if conanfile:  # FIXME: Remove when deprecated win.py msbuild tools
```

Comment:
I don't get why this ``if`` is there. If False, the rest of the MSBuild will fail, won't it?

Reply:
Use the typical condition or try-except
- If it is a conanfile do one thing, if it is settings do other thing.